src/main/java/DaveTheBrave.java

```diff
@@ -0,0 +1,369 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Paths;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.List;
+import java.util.ArrayList;
+import java.time.*;
+
+class Task {
+    private String type; // "T" = To-Do, "D" = Deadline, "E" = Events
+    private String task;
+    private boolean status; // false = not done, true = done
+    private String details; // Info for Event tasks
+    private LocalDate deadline; // Info for Deadline tasks
+
+    public static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    public static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy");
+
+    public Task(String type, String task, String details) {
+        this.type = type;
+        this.task = task;
+        this.status = false;
+        this.details = details;
+
+        if (type.equals("D") && details != null) {
+            try {
+                this.deadline = LocalDate.parse(details.trim(), INPUT_FORMATTER);
+            } catch (DateTimeParseException e) {
+                System.out.println("Invalid date format! Please use yyyy-MM-dd (e.g., 2019-10-15).");
+            }
+        }
+    }
+
+    public void mark() {
+        this.status = true;
+    }
+
+    public void unmark() {
+        this.status = false;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public String getTask() {
+        return task;
+    }
+
+    public boolean getStatus() {
+        return status;
+    }
+
+    public String getDetails() {
+        return details;
+    }
+
+    @Override
+    public String toString() {
+        String taskStatus = status ? "[X]" : "[ ]";
+        if (type.equals("D")) {
+            String taskDetails = details != null ? " (by: " + deadline.format(OUTPUT_FORMATTER) + ")" : "";
+            return "[" + type + "]" + taskStatus + " " + task + taskDetails;
+        } else {
+            String taskDetails = details != null ? " " + details : "";
+            return "[" + type + "]" + taskStatus + " " + task + (taskDetails);
+        }
+    }
+
+    // Writing to file
+    public String toFileFormat() {
+        return type + " | " + (status ? "[X]" : "[ ]") + " | " + task + (details != null ? " | " + details : "");
+    }
+
+    // Loading from file
```

Comment:
Use Javadoc comment format. Intellij IDE has this feature. Type /** and enter. The IDE will create the comment skeleton.